Added ruleset for structs vs classes

Haud opened the pull request:
@ThibaultKlein @paulmiard

Haud committed:
Added ruleset for structs vs classes

ThibaultKlein commented on README.md:
```diff
@@ -172,6 +173,19 @@ class Object {
+or where equality does not make sense, favor `class`.
+
+In general, you should favor structs and protocols over classes; even in cases where polymorphism would dictate the usage of a class, consider if you can
+achieve a similar result via protocols and extensions.
```
if we could mention the keyword `composition` versus `inheritance` it would make me happy ;)

ThibaultKlein commented:
it would be good to have bullet points to make it easier to read, currently I feel that the information is hidden in the paragraphs, we should make it very easy to find information in the README

Haud commented:
@ThibaultKlein updated based on your feedback. I moved the basic points to bullet points, added some more tangible benefits for struct-usage, and mentioned composition and inheritance JUST FOR YOU !

Haud commented:
Also paging @paulmiard

ThibaultKlein commented on README.md:
```diff
@@ -172,6 +173,28 @@ class Object {
+
+* immutable
+* stateless
+* have a definition for equality
```
Capitalize?

Haud replied:
done

ThibaultKlein commented:
:+1:
